script.js

```diff
@@ -1,29 +1,42 @@
 //local storage//
-const LOCAL_STORAGE_KEY = "user_data";
-
-//fetching data //
-let promise= fetch("https://geektrust.s3-ap-southeast-1.amazonaws.com/adminui-problem/members.json");
-promise.then((response)=>{
+localStorage.removeItem('userdata');
+const UserData = "userdata";
+
+//fetching data and handling local storage//
+//import APIurl from config.js
+import { APIurl } from "./config.js"; //import api url
+let p= fetch(APIurl);
+p.then((response)=>{
     if(!response.ok)
         throw new Error("Network response is not ok"+response.statusText);
     else
 return response.json();
 }).then((data)=>{
     // Check if there's data in local storage
-    let storedData = localStorage.getItem(LOCAL_STORAGE_KEY);
+    let storedData = localStorage.getItem(UserData);
     if (storedData) {
         data = JSON.parse(storedData); // Use data from local storage if available
     } else {
-        localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(data)); // Store fetched data in local storage
+        localStorage.setItem(UserData, JSON.stringify(data)); // Store fetched data in local storage
     }
-    console.log(data);
-
+    adminUI(data);
+})
+.catch((error)=>{
+    console.error("There is an error: ", error);
+});
 
-    //Since search, edit, delete, pagination, etc., 
-    //typically depend on the fetched data, these actions should be handled inside the .then() block.
-    //lets create the rows in table//
+ //this function contains all DOM_related code
+function adminUI(data){
 const tablebody = document.querySelector(".tbody");
+const searchbutton =document.querySelector(".searchbtn");
+const searchText= document.querySelector(".search");
+const selectALLcheckbox= document.querySelector(".selectAll");
+const firstPage = document.querySelector("#firstpage");
+const previousPage = document.querySelector("#previouspage");
+const nextPage = document.querySelector("#nextpage");
+const lastPage = document.querySelector("#lastpage");
 
+//create the rows in table//
 const createRows =(data)=>{
```

Comment:
why have you created the function inside a function, any reason? can you create the createRows function outside the function?